solidity/test/zeto_anon_enc.ts

```diff
@@ -51,10 +75,77 @@ describe("Zeto based fungible token with anonymity and encryption", function ()
     Bob = await newUser(b);
     Charlie = await newUser(c);
 
-    ({ deployer, zeto, erc20 } = await deployZeto('Zeto_AnonEnc'));
+    ({ deployer, zeto, erc20 } = await deployZeto("Zeto_AnonEnc"));
 
-    circuit = await loadCircuit('anon_enc');
-    ({ provingKeyFile: provingKey } = loadProvingKeys('anon_enc'));
+    circuit = await loadCircuit("anon_enc");
+    ({ provingKeyFile: provingKey } = loadProvingKeys("anon_enc"));
+    batchCircuit = await loadCircuit("anon_enc_batch");
+    ({ provingKeyFile: batchProvingKey } = loadProvingKeys("anon_enc_batch"));
+  });
+
+  it("(batch) mint to Alice and batch transfer 10 UTXOs honestly to Bob should succeed", async function () {
+    // first mint the tokens for batch testing
+    const inputUtxos = [];
+    for (let i = 0; i < 10; i++) {
+      // mint 10 utxos
+      inputUtxos.push(newUTXO(1, Alice));
+    }
+    await doMint(zeto, deployer, inputUtxos);
+
+    // Alice proposes the output UTXOs, 1 utxo to bob, 2 utxos to alice
+    const _bOut1 = newUTXO(8, Bob);
+    const _bOut2 = newUTXO(1, Alice);
+    const _bOut3 = newUTXO(1, Alice);
+    const outputUtxos = [_bOut1, _bOut2, _bOut3];
+    const outputOwners = [Bob, Alice, Alice];
+    const inflatedOutputUtxos = [...outputUtxos];
+    const inflatedOutputOwners = [...outputOwners];
+    for (let i = 0; i < 10 - outputUtxos.length; i++) {
+      inflatedOutputUtxos.push(ZERO_UTXO);
+      inflatedOutputOwners.push(Bob);
+    }
+
+    // Alice transfers UTXOs to Bob
+    const result = await doTransfer(
+      Alice,
+      inputUtxos,
+      inflatedOutputUtxos,
+      inflatedOutputOwners,
+    );
+
+    const events = parseUTXOEvents(zeto, result.txResult!);
+    expect(events[0].inputs).to.deep.equal(inputUtxos.map((i) => i.hash));
+    const incomingUTXOs: any = events[0].outputs;
+    const ecdhPublicKey = events[0].ecdhPublicKey;
+    // Bob reconstructs the shared key using his private key and ephemeral public key
+
+    const sharedKey = genEcdhSharedKey(Bob.babyJubPrivateKey, ecdhPublicKey);
+    const plainText = poseidonDecrypt(
+      events[0].encryptedValues,
+      sharedKey,
+      events[0].encryptionNonce,
+      20,
```

Comment:
related to the comment against the circom code on encryption, this test code should be evident that the behavior here is undesirable: Bob is able to decrypt the secrets for all the outputs, even though (usually) only one of the UTXOs are for him

Reply:
the expected behavior should be that the encrypted values are groups of 4's, with each 4-element slice representing the 2 encrypted values (value + salt) for each output receiver. Bob can attempt to decrypt them all, but only one attempt would succeed, with the others failing the decryption call due to the parity bits (which are the extra 2 values).

Reply:
Got it. I'll update the circuit to output an array of public keys and an array of cipher text for each individual output utxos